packages/app/src/cli/services/bulk-operations/stage-file.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  StagedUploadsCreate,
+  StagedUploadsCreateMutation,
+  StagedUploadsCreateMutationVariables,
+} from '../../api/graphql/bulk-operations/generated/staged-uploads-create.js'
+import {adminRequestDoc} from '@shopify/cli-kit/node/api/admin'
+import {AdminSession} from '@shopify/cli-kit/node/session'
+import {readFile, fileSize} from '@shopify/cli-kit/node/fs'
+import {basename} from '@shopify/cli-kit/node/path'
+import {formData, fetch} from '@shopify/cli-kit/node/http'
+import {AbortError} from '@shopify/cli-kit/node/error'
+
+interface StageFileOptions {
+  adminSession: AdminSession
+  filePath?: string
+  jsonVariables?: string
+  filename?: string
+}
+
+export async function stageFile(options: StageFileOptions): Promise<string> {
```

Comment:
This is a really long function -- can we think of ways to break it up into a bunch of small functions to make it easier to read?